runtime/lua/vim/health.lua

```diff
@@ -331,13 +349,30 @@ function M._check(mods, plugin_names)
 
   local emptybuf = vim.fn.bufnr('$') == 1 and vim.fn.getline(1) == '' and 1 == vim.fn.line('$')
 
-  -- When no command modifiers are used:
-  -- - If the current buffer is empty, open healthcheck directly.
-  -- - If not specified otherwise open healthcheck in a tab.
-  local buf_cmd = #mods > 0 and (mods .. ' sbuffer') or emptybuf and 'buffer' or 'tab sbuffer'
-
   local bufnr = vim.api.nvim_create_buf(true, true)
-  vim.cmd(buf_cmd .. ' ' .. bufnr)
+  if
+    vim.g.health
+    and type(vim.g.health) == 'table'
+    and vim.tbl_get(vim.g.health, 'style') == 'float'
+  then
+    local syntax = vim.tbl_get(vim.g.health, 'syntax') or ''
+    local float_bufnr, float_winid = vim.lsp.util.open_floating_preview({}, syntax, {
+      height = math.floor(vim.o.lines * 0.8),
+      width = math.floor(vim.o.columns * 0.8),
+      border = vim.tbl_get(vim.g.health, 'border'),
```

Comment:
redundant ? vim.lsp.util.open_floating_preview.Opts have it, but we will have border option so can instead there i guess..